src/datasets/info.py

```diff
@@ -166,6 +168,13 @@ def __post_init__(self):
             else:
                 template = task_template_from_dict(self.task_templates)
                 self.task_templates = [template] if template is not None else []
+            # Insert labels and mappings for text classification
+            for idx, template in enumerate(self.task_templates):
+                if isinstance(template, TextClassification) and self.features is not None:
+                    # Cast labels to tuple to enable hashing of task template
+                    object.__setattr__(template, "labels", tuple(sorted(self.features[template.label_column].names)))
```

Comment:
Maybe raise an error if `template.labels` is not None and and doesn't match the labels in info.features ?

If users want to create a custom TaskTemplate with custom labels, then they must match the dataset labels.
e.g. this should fail:

```python
ds = load_dataset("emotion")
labels = ['label_that_doesnt_exist', 'another_one_that_doesnt_exist']
tc = TextClassification(text_column="text", label_column="label", labels=labels)
ds.prepare_for_task(tc)
```

Reply:
> Maybe raise an error if `template.labels` is not None and and doesn't match the labels in info.features ?

Great idea! I decided to raise the error in `prepare_for_task` because `template.labels` is `None` by default in `DatasetInfo.__post_init__` and is the reason we set the attribute there with:

```python
# Cast labels to tuple to enable hashing of task template                   
 template.__dict__["labels"] = tuple(sorted(self.features[template.label_column].names))
```

I've also added a test to verify that we raise the error as expected (I also discovered some of my previous assertion tests were passing for the wrong reason 😅 ).

Perhaps one drawback from this approach is that we force the user to explicitly define the labels if they want to create their own `TextClassification` instance. On the other hand, being explicit is the Python way 😄 

> Also maybe we should create another instance of `TextClassification`, so that the user could reuse a TaskTemplate with unset labels later, instead of updating the current object.

I think that by raising the error in `prepare_for_task`, we no longer need a new instance because an instance with unset labels will raise a `ValueError` and not be modified. Please correct me if I misunderstood your remark.

Reply:
Thanks :) Let's allow labels to be None and sync with the features when appropriate (currently in DatasetInfo.__post_init__)

Regarding the new instance: I created a new instance in the changes I pushed.

Looks all good now